README.md

```diff
@@ -1,3 +1,262 @@
+
+### Writing to Bigtable
+
+You can use the `dataFrame.write()` format write to Bigtable.
+This is a sample snippet of writing to Bigtable using Java:
+
+```java
+Dataset<Row> dataFrame;
+// Adding some values to dataFrame.
+dataFrame
+  .write()
+  .format("bigtable")
+  .option("catalog", catalog)
+  .option("spark.bigtable.project.id", projectId)
+  .option("spark.bigtable.instance.id", instanceId);
+```
+
+### Reading from Bigtable
+
+You can use the `spark.read()` format to read from Bigtable.
+This is a sample snippet of reading from Bigtable using Java:
+
+```java
+Dataset<Row> dataFrame = spark
+    .read()
+    .format("bigtable")
+    .option("catalog", catalog)
+    .option("spark.bigtable.project.id", projectId)
+    .option("spark.bigtable.instance.id", instanceId)
+    .load();
+```
+
+### Runtime configurations
+
+You can use `.option(<config_name>, <config_value>)` in Spark to pass different
+runtime configurations to
+the connector. For example, Bigtable project and instance ID or settings for
+timestamp and timeout configurations.
+For a full list of configurations, refer to
+`spark-bigtable_2.12/src/main/scala/com/google/cloud/spark/bigtable/datasources/BigtableSparkConf.scala`,
+where these configs are defined.
+
+### Bigtable emulator support
+
+When using the connector locally, you can start a Bigtable emulator server and
+set the environment variable
+`export BIGTABLE_EMULATOR_HOST=localhost:<emulator_port>` to use the connector
+with the emulator instead of a
+real Bigtable instance. You can refer to the
+[Bigtable emulator documentations](https://cloud.google.com/bigtable/docs/emulator)
+for more details on using it.
+
+### Complex data type serialization using Apache Avro
+
+You can specify an Avro schema for columns with a complex Spark SQL type such as
+`ArrayType`, `MapType`, or `StructType`,
+to serialize and store them in Bigtable.
+
+### Row key filter push down
+
+This connector supports pushing down some of the filters on the row key column
+in the DataFrame to Bigtable
+and performing them on the server-side. The list of supported or non-supported
+filters is as follows:
+
+| Filter               | Push down filter supported |
+|----------------------|----------------------------|
+| `EqualTo`            | Yes                        |
+| `LessThan`           | Yes                        |
+| `GreaterThan`        | Yes                        |
+| `LessThanOrEqual`    | Yes                        |
+| `GreaterThanOrEqual` | Yes                        |
+| `StringStartsWith`   | Yes                        |
+| `Or`                 | Yes                        |
+| `And`                | Yes                        |
+| `Not`                | No                         |
+| Compound Row Key     | No                         |
+
+Note that when using compound row keys, filter on those columns are
+**not** pushed to Bigtable and are performed on the client-side (resulting in a
+full-table scan).
```

Comment:
do we have any references to what to use instead of compound row keys? Is it possible to reference that here. 

e.g. Instead, if filtering is required, use a logical row key to represent the entire row as a single column and one of the above supported filters.

Reply:
I added a small note about how to concatenate columns into a new one, ptal